doc/image_proc.rst

```diff
@@ -0,0 +1,344 @@
+e-/ADU) for each amplifier.  Even though :math:`O` and :math:`B` are in ADU in
+the equation for :math:`p` above, they can be measured in e- by applying the
+gain.  More importantly, how they are measured must be consistent across all
+images.  The ``apply_gain`` parameter defaults to true, and you should rarely
+(if ever) want to change this.  However, if you do, **make sure you change it
+for all images you process**; i.e., you don't want to be bias subtracting an
+image in ADU (``apply_gain = False``) from an image in e- (``apply_gain =
+True``).  The units of the processed images are saved to the image header as
+``UNITS``, which can be either ``'e-'`` or ``'ADU'``.
+
+Pattern-Noise Subtraction
+-------------------------
+
+.. Ryan, please check this.
+
+Some instruments, specifically Keck KCWI, are known to have a sinusoidal pattern
+in its bias.  :func:`~pypeit.images.rawimage.RawImage.subtract_pattern` models
+and subtracts this pattern from the data based on the overscan regions.  Unless
+you know such a pattern exists in your data, you should set the ``use_pattern``
+option to false (the default) for *all* frames.  Currently no error associated
+with this pattern subtraction is included in the image-processing error budget
+(unless the readnoise is determined empirically; see below).
+
+Read and Digitization Noise
+---------------------------
+
+Readnoise variance, :math:`V_{\rm rn}`, is calculated by
+:func:`~pypeit.core.procimg.rn2_frame`.  The calculation requires the detector
+readnoise (RN in elections, e-) and, possibly, gain (:math:`\gamma` in e-/ADU)
+for each amplifier, which are provided for each amplifier by the
+:class:`~pypeit.images.detector_container.DetectorContainer` object defined for
+each detector in each :class:`~pypeit.spectrographs.spectrograph.Spectrograph`
+subclass.  If a readnoise value is set to :math:`\leq 0`, the readnoise is
+estimated by the variance in the overscan regions of the image being processed;
+see :func:`~pypeit.images.rawimage.RawImage.estimate_readnoise`; use of the
+readnoise estimate regardless of the value provided by the
+:class:`~pypeit.images.detector_container.DetectorContainer` object can also be
+explicitly requested by setting the ``empirical_rn`` parameter to true.
+
+By default, the readnoise variance image does *not* include digitization noise.
+Digitization noise is driven by the conversion from counts to ADU via the gain
+and the quantization to an integer.  One can be derive the digitization noise by
+taking the second moment of a uniform distribution from -1/2 to 1/2 to find
+:math:`\sqrt{1/12}` ADU [1]_ [2]_, which is typically negligible (i.e., when the
+gain is on the same order as the readnoise).  And, in most cases, digitization
+noise will have been included in the estimate of the readnoise.  For this
+reason, digitization noise is *not* explicitly included in the ``PypeIt``
+variance model, and its inclusion cannot be turned on using a pypeit parameter.
```

Comment:
put this pypeit in `pypeit`

Reply:
Done.  (If I understand what you meant.)